components/playlist/browser/playlist_download_request_manager.cc

```diff
@@ -296,7 +301,8 @@ void PlaylistDownloadRequestManager::ProcessFoundMedia(
 
     auto item = mojom::PlaylistItem::New();
     item->id = base::Token::CreateRandom().ToString();
-    item->page_source = GURL(*page_source);
+    item->page_source = requested_url;
```

Comment:
@deeppandya , I changed this here, so you might be able to iterate through to see if something has been added to a page .